src/plots/plots.js

```diff
@@ -1876,11 +1878,9 @@ plots.doAutoMargin = function(gd) {
                     var pr = pushMargin[k2].r.size;
 
                     if(fr > fl) {
-                        var newl = (pl * fr +
-                            (pr - fullLayout.width) * fl) / (fr - fl);
-                        var newr = (pr * (1 - fl) +
-                            (pl - fullLayout.width) * (1 - fr)) / (fr - fl);
-                        if(newl >= 0 && newr >= 0 && newl + newr > ml + mr) {
+                        var newl = (pl * fr + (pr - width) * fl) / (fr - fl);
```

Comment:
It may be more readable for some people if `camelCase` is used with new variables: `newl` > `newL`

Reply:
Good call done in https://github.com/plotly/plotly.js/pull/3605/commits/d5bacfefbc0d75cfaf81a70a568bc27e08ed2c30